controllers/reconcile.go

```diff
@@ -1211,6 +1232,45 @@ func (r *FrontendReconciliation) setupBundleData(_ *v1.ConfigMap, _ map[string]c
 	return nil
 }
 
+func setupCaddyFile() string {
+	caddyFileContent := `
+	{
+		{$CADDY_TLS_MODE}
```

Comment:
Ideally I'd like to have this read in from a file via `embed` or something similar https://pkg.go.dev/embed

Would be nice if all this config was just on the `FrontendEnvironment` so we could update it at run time but that might not pass the cost / benefit test. Would mean we could update the whole frontend fleet's config without a release. But it would also mean either storing this whole config as a big old string in a YAML resource (not impossible but icky) OR actually pulling in the caddy file spec or something and popping that onto the FE.

Would be worth getting @psav's opinion.